File: leetcode/p594.py
class Solution:
    def findLHS(self, nums: List[int]) -> int:
        ns = Counter(nums)
        max_len = 0
        for i in ns:
            if i+1 in ns:
                max_len = max(max_len, ns[i]+ns[i+1])
        return max_len

File: leetcode/p594.cpp
class Solution {
public:
    int findLHS(vector<int>& nums) {
        unordered_map<int,int> counter;
        int max_len = 0;
        for (int i=0;i<nums.size();i++) {
            counter[nums[i]]++;
        }
        for (auto it: counter) {
            int t = it.first;
            if (counter.find(t+1) != counter.end() && counter[t]+counter[t+1]>max_len) {
                max_len = counter[t]+counter[t+1];
            }
        }
        return max_len;
    }
};
